PR for SRVLOGIC-189: Added the installtion and uninstallation assemblies for Openshift Serverless Logic docs (#76488)
openshift/openshift-docs: modules/serverless-logic-install-kn-workflow-artifact-images.adoc

Reviewer: The commands in this procedure should be `[source,terminal]`

Author: fixed everywhere

// Module included in the following assemblies:
//
// * /serverless/install/serverless-logic-install-kn-workflow-plugin-cli.adoc

:_content-type: PROCEDURE
[id="serverless-logic-install-kn-workflow-artifact-images_{context}"]
= Installing the OpenShift Serverless Logic Knative Workflow plugin using the artifacts image

.Prerequisites

* You have installed the Knative (`kn`) CLI.
* You have installed Podman on your local machine. 

.Procedure

. Download the Knative Workflow plugin using the `logic-kn-workflow-cli-artifacts-rhel8` image by running the following commands:

> 

Reviewer 2: ```suggestion

.. Use the following command from a system with podman installed
+
[source, shell]
----
podman login registry.redhat.io

Username: myrhusername
Password:  *****************
Login Succeeded!
----

You can use your Red Hat Customer Portal user and password, or a Registry Service Account username and password.

```

Author: Added this step

// Module included in the following assemblies:
//
// * /serverless/install/serverless-logic-install-kn-workflow-plugin-cli.adoc

:_content-type: PROCEDURE
[id="serverless-logic-install-kn-workflow-artifact-images_{context}"]
= Installing the OpenShift Serverless Logic Knative Workflow plugin using the artifacts image
:context: serverless-logic-install-kn-workflow-artifact-images

.Prerequisites

> * You have installed the Knative (`kn`) CLI.

Reviewer 2: ```suggestion
* You have installed Podman in your local machine.
```

Reviewer 2: on the other hand we need Podman because we'll execute podman commands

> * You have access to a project with the appropriate roles and permissions to create applications and other workloads in {ocp-product-title}.

Reviewer 2: IDK what is the standard you use here in general, but, from the perspective of the solely installation of the kn-workflow plugin with this new procedure, you don't access to any project. Users can even install the plugin to "play" locally and never connect to the OpenShift

Author: removed this prereq

> * You have access to a project with the appropriate roles and permissions to create applications and other workloads in {ocp-product-title}.

Reviewer 2: ```suggestion
```

.Procedure

. Download the Knative Workflow plugin using the `logic-kn-workflow-cli-artifacts-rhel8` image by running the following commands:

.. Start the `logic-kn-workflow-cli-artifacts-rhel8` image.
+
[source,shell]
----
$ export KN_IMAGE=registry.redhat.io/openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0
----
+
[source,shell]
----
$ export KN_CONTAINER_ID=$(podman run -di $KN_IMAGE)
----

.. Copy the Knative Workflow plugin binary file according to your environment.
+
.Binary file for Linux amd64 architecture
[source,shell]
----
$ podman cp $KN_CONTAINER_ID:/usr/share/kn/linux_amd64/kn-workflow-linux-amd64.tar.gz kn-workflow-linux-amd64.tar.gz
----
+
.Binary file for Linux arm64 architecture
[source,shell]
----
$ podman cp $KN_CONTAINER_ID:/usr/share/kn/linux_arm64/kn-workflow-linux-arm64.tar.gz kn-workflow-linux-arm64.tar.gz
----
+

> .Binary file for macOS amd64 arm64 architecture

Reviewer 2: ```suggestion
.Binary file for macOS amd64 architecture
```

> .Binary file for macOS amd64 arm64 architecture

Reviewer 2: ```suggestion
.Binary file for macOS amd64 architecture
```

// Module included in the following assemblies:
//
// * /serverless/install/serverless-logic-install-kn-workflow-plugin-cli.adoc

:_content-type: PROCEDURE
[id="serverless-logic-install-kn-workflow-artifact-images_{context}"]
= Installing the OpenShift Serverless Logic Knative Workflow plugin using the artifacts image

.Prerequisites

* You have installed the Knative (`kn`) CLI.
* You have installed Podman on your local machine. 

.Procedure

. Download the Knative Workflow plugin using the `logic-kn-workflow-cli-artifacts-rhel8` image by running the following commands:

.. To login to the Red{nbsp}Hat Registry run the following command on a system with podman installed:
+
[source,shell]
----
$ podman login registry.redhat.io
----
+

> You can use your Red{nbsp}Hat customer portal or a registry service account username and password.

Reviewer: ```suggestion
You can use your Red{nbsp}Hat Customer Portal account or a registry service account.
```

> .. Start the `logic-kn-workflow-cli-artifacts-rhel8` image.
> +
> [source,shell]
> ----
> $ export KN_IMAGE=registry.redhat.io/openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0
> ----
> +
> [source,shell]
> ----
> $ export KN_CONTAINER_ID=$(podman run -di $KN_IMAGE)
> ----

Reviewer: this appears to be two separate steps for setting two variables.

```suggestion
.. Set a variable for [whatever this is] by running the following command:
+
[source,shell]
----
$ export KN_IMAGE=registry.redhat.io/openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0
----

.. Set a variable for [whatever this is] by running the following command:
+
[source,shell]
----
$ export KN_CONTAINER_ID=$(podman run -di $KN_IMAGE)
----
```

Author: Alright! I have added this

> .. Copy the Knative Workflow plugin binary file according to your environment.
> +
> .Binary file for Linux amd64 architecture
> [source,shell]
> ----
> $ podman cp $KN_CONTAINER_ID:/usr/share/kn/linux_amd64/kn-workflow-linux-amd64.tar.gz kn-workflow-linux-amd64.tar.gz
> ----
> +
> .Binary file for Linux arm64 architecture
> [source,shell]
> ----
> $ podman cp $KN_CONTAINER_ID:/usr/share/kn/linux_arm64/kn-workflow-linux-arm64.tar.gz kn-workflow-linux-arm64.tar.gz
> ----
> +
> .Binary file for macOS amd64 architecture
> [source,shell]
> ----
> $ podman cp $KN_CONTAINER_ID:/usr/share/kn/macos_amd64/kn-workflow-macos-amd64.tar.gz kn-workflow-macos-amd64.tar.gz
> ----
> +
> .Binary file for macOS arm64 architecture
> [source,shell]
> ----
> $ podman cp $KN_CONTAINER_ID:/usr/share/kn/macos_arm64/kn-workflow-macos-arm64.tar.gz kn-workflow-macos-arm64.tar.gz
> ----
> +
> .Binary file for Windows amd64 architecture
> [source,shell]
> ----
> $ podman cp $KN_CONTAINER_ID:/usr/share/kn/windows/kn-workflow-windows-amd64.zip kn-workflow-windows-amd64.zip
> ----

Reviewer: In this step, change `[source,shell` to `[source,terminal]`. Optionally, you might consider a more typical format, as follows.

Usually, when a command has multiple options like this, we include the command a single time and explain the options as required. In this case, the source file and the target filename are the same, so you can also replace the repeated filename at the end with a `.`

In this case, something like:
```suggestion
.. Copy the Knative Workflow plugin binary file by running the following command:
+
[source,terminal]
----
$ podman cp $KN_CONTAINER_ID:/usr/share/kn/<path_to_binary> .
----
+
where `<path_to_binary>` is the path to the file for your environment:
+
[cols=2,options=header]
|===
|Environment
|Path to binary file

|Linux amd64 architecture
|`linux_amd64/kn-workflow-linux-amd64.tar.gz`

|Linux arm64 architecture
|`linux_arm64/kn-workflow-linux-arm64.tar.gz`

|macOS amd64 architecture
|`macos_amd64/kn-workflow-macos-amd64.tar.gz`

|macOS arm64 architecture
|`macos_arm64/kn-workflow-macos-arm64.tar.gz`

|Windows amd64 architecture
|`windows/kn-workflow-windows-amd64.zip`

|===
```

If you're concerned about using a partial path, you could do this:

```suggestion
.. Copy the Knative Workflow plugin binary file by running the following command:
+
[source,terminal]
----
$ podman cp $KN_CONTAINER_ID:<path_to_binary> .
----
+
where `<path_to_binary>` is the path to the file for your environment:
+
[cols=2,options=header]
|===
|Environment
|Path to binary file

|Linux amd64 architecture
|`/usr/share/kn/linux_amd64/kn-workflow-linux-amd64.tar.gz`

|Linux arm64 architecture
|`/usr/share/kn/linux_arm64/kn-workflow-linux-arm64.tar.gz`

|macOS amd64 architecture
|`/usr/share/kn/macos_amd64/kn-workflow-macos-amd64.tar.gz`

|macOS arm64 architecture
|`/usr/share/kn/macos_arm64/kn-workflow-macos-arm64.tar.gz`

|Windows amd64 architecture
|`/usr/share/kn/windows/kn-workflow-windows-amd64.zip`

|===
```

> .. Stop the container by running the following commands:

Reviewer: Only one command per step
```suggestion
.. Stop the container by running the following command:
```

+
[source,shell]
----
$ podman stop $KN_CONTAINER_ID
----

> +
> [source,shell]
> ----
> $ podman rm $KN_CONTAINER_ID
> ----

Reviewer: This is a separate command and needs its own step. I think probably:
```suggestion
.. Delete the container by running the following command:
+
[source,shell]
----
$ podman rm $KN_CONTAINER_ID
----
```

> .. Extract the selected Knative Workflow plugin binary file and rename it to `kn-workflow` by running the following commands:

Reviewer: One command per step.
```suggestion
.. Extract the selected Knative Workflow plugin binary file by running the following command:
```

+
[source,shell]
----
$ tar xvzf kn-workflow-linux-amd64.tar.gz
----

> +
> [source,shell]
> ----
> $ mv kn kn-workflow
> ----

Reviewer: This is a second step
```suggestion
Rename the Knative Workflow plugin binary file to `kn-workflow` by running the following command:
+
[source,terminal]
----
$ mv kn kn-workflow
----
```

+
[source,terminal]
----
$ podman rm $KN_CONTAINER_ID
----

.. Extract the selected Knative Workflow plugin binary file by running the following command:
+
[source,terminal]
----
$ tar xvzf kn-workflow-linux-amd64.tar.gz
----

.. Rename the Knative Workflow plugin binary file to `kn-workflow` by running the following command:
+
[source,terminal]
----
$ mv kn kn-workflow
----

. Install the `kn-workflow` command as a plugin of the Knative CLI by running the following commands:

.. Copy the `kn-workflow` binary file to a directory in your PATH, such as `/usr/local/bin`, and ensure the file name is `kn-workflow`:
+
[source,terminal]
----
$ cp path/to/downloaded/kn-workflow /usr/local/bin/kn-workflow
----

.. Make the binary file executable:
+
[source,terminal]
----
$ chmod +x /usr/local/bin/kn-workflow
----

.Verification
. Run the following command to verify that the `kn-workflow` plugin is installed successfully:
+
[source,terminal]
----
$ kn plugin list
----

. After installing the plugin, you can use `kn-workflow` to run the related subcommands.
+
.Aliases to use workflow subcommand
[source,terminal]
----
kn-workflow
----
+

> .Example output of `kn-workflow`

Reviewer 2: ```suggestion
.Example output of `kn-workflow help` command
```
